bigframes/pandas/io/api.py

```diff
@@ -263,11 +264,13 @@ def _read_gbq_colab(
     if pyformat_args is None:
         pyformat_args = {}
 
-    query = bigframes.core.pyformat.pyformat(
+    create_query = functools.partial(
+        bigframes.core.pyformat.pyformat,
         query_or_table,
         pyformat_args=pyformat_args,
+        dry_run=True,
     )
-    _set_default_session_location_if_possible(query)
+    _set_default_session_location_if_possible_deferred_query(create_query)
```

Comment:
If it's a dry run, maybe we don't even try to run this? I'd like to delay selecting a location until we run a real query.

Reply:
Done.